meinberlin/apps/projects/insights.py

```diff
@@ -0,0 +1,118 @@
+from typing import Iterable
+from typing import List
+
+from django.core.exceptions import FieldDoesNotExist
+from django.db.models import Q
+
+from adhocracy4.comments.models import Comment
+from adhocracy4.polls.models import Answer
+from adhocracy4.polls.models import Poll
+from adhocracy4.polls.models import Vote
+from adhocracy4.projects.models import Project
+from adhocracy4.ratings.models import Rating
+from meinberlin.apps.budgeting.models import Proposal
+from meinberlin.apps.ideas.models import Idea
+from meinberlin.apps.likes.models import Like
+from meinberlin.apps.livequestions.models import LiveQuestion
+from meinberlin.apps.mapideas.models import MapIdea
+from meinberlin.apps.maptopicprio.models import MapTopic
+from meinberlin.apps.projects.models import ProjectInsight
+from meinberlin.apps.topicprio.models import Topic
+
+
+def create_insights(projects: Iterable[Project]) -> List[ProjectInsight]:
+    return [create_insight(project=project) for project in projects]
+
+
+def create_insight(project: Project) -> ProjectInsight:
+    modules = project.modules.all()
+
+    ideas = Idea.objects.filter(module__in=modules)
+    map_ideas = MapIdea.objects.filter(module__in=modules)
+    comments = Comment.objects.filter(
+        Q(project=project) | Q(parent_comment__project=project)
+    )
+    proposals = Proposal.objects.filter(module__in=modules)
+    polls = Poll.objects.filter(module__in=modules)
+    votes = Vote.objects.filter(choice__question__poll__in=polls)
+    answers = Answer.objects.filter(question__poll__in=polls)
+    live_questions = LiveQuestion.objects.filter(module__in=modules)
+    likes = Like.objects.filter(question__in=live_questions)
+    topics = Topic.objects.filter(module__in=modules)
+    map_topics = MapTopic.objects.filter(module__in=modules)
+
+    values = [Rating.POSITIVE, Rating.NEGATIVE]
+    ratings_ideas = Rating.objects.filter(idea__in=ideas, value__in=values)
+    ratings_map_ideas = Rating.objects.filter(mapidea__in=map_ideas, value__in=values)
+    ratings_comments = Rating.objects.filter(comment__in=comments, value__in=values)
+    ratings_topics = Rating.objects.filter(topic__in=topics, value__in=values)
+    ratings_map_topics = Rating.objects.filter(
+        maptopic__in=map_topics, value__in=values
+    )
```

Comment:
what about the plans? @goapunk @vellip

Reply:
I think, since plans are hierarchy-wise above projects, we don't need to track them, right?